pkg/controller/controller_instance.go

```diff
@@ -1434,8 +1434,16 @@ func (c *controller) processProvisionSuccess(instance *v1beta1.ServiceInstance,
 	setServiceInstanceDashboardURL(instance, dashboardURL)
 	setServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionReady, v1beta1.ConditionTrue, successProvisionReason, successProvisionMessage)
 	instance.Status.ExternalProperties = instance.Status.InProgressProperties
+	currentReconciledGeneration := instance.Status.ReconciledGeneration
 	clearServiceInstanceCurrentOperation(instance)
 
+	if instance.DeletionTimestamp != nil {
```

Comment:
I think this is a fine workaround for now but I do think that we should consider, as a follow-up to this, storing the generation that the in-progress operation is for, so that when we complete an operation we can set the reconciled generation correctly to the generation that started the operation.

Reply:
Agreed, we should store a correct `reconciledGeneration` 😀

Reply:
@jboyd01 leave a TODO then?